Use the correct form of sha1.Sum

mcsaucy opened the pull request:
sha1.New().Sum(x) appends a hash of nothing to x.
See https://go.dev/play/p/h6pGFHEzbfA

Interestingly, this didn't break the corruption check because the generated values were still unique as the code compared the raw values. They're just not hashes.

derekcollison commented:
https://app.travis-ci.com/github/nats-io/graft/jobs/632027526

derekcollison reviewed (commented):
So if someone uses the lib and has some written state, will this change break their system?

mcsaucy commented:
> So if someone uses the lib and has some written state, will this change break their system?

Added fallback logic.

> Also be good to have a test that checks that the sha is correct.

Added a test case for the fallback logic. I'm not sure what you want me to test here, so LMK if you want more.

mcsaucy commented:
> For the fix, have it write the sha but then check it directly from disk etc..

Done.

derekcollison commented:
Need to fix up CI failures.

https://app.travis-ci.com/github/nats-io/graft/jobs/632199047

mcsaucy commented:
The CI failures are staticcheck fails because the RPC mechanism uses the deprecated nats.EncodedConn. Fixing that feels out of scope for this change to state persistence.

```
$ if [[ "$TRAVIS_GO_VERSION" =~ 1.24 ]]; then find . -type f -name "*.go" | xargs misspell -error -locale US; staticcheck ./...; fi
nats_rpc.go:45:6: nats.EncodedConn is deprecated: Encoded connections are no longer supported.  (SA1019)
nats_rpc.go:67:13: nats.NewEncodedConn is deprecated: Encoded connections are no longer supported.  (SA1019)
nats_rpc.go:76:13: nats.NewEncodedConn is deprecated: Encoded connections are no longer supported.  (SA1019)
nats_rpc.go:89:19: rpc.ec.Subscribe is deprecated: Encoded connections are no longer supported.  (SA1019)
nats_rpc.go:94:21: rpc.ec.Subscribe is deprecated: Encoded connections are no longer supported.  (SA1019)
nats_rpc.go:120:3: rpc.ec.Close is deprecated: Encoded connections are no longer supported.  (SA1019)
nats_rpc.go:170:14: rpc.ec.Subscribe is deprecated: Encoded connections are no longer supported.  (SA1019)
nats_rpc.go:182:9: rpc.ec.PublishRequest is deprecated: Encoded connections are no longer supported.  (SA1019)
nats_rpc.go:194:9: rpc.ec.Publish is deprecated: Encoded connections are no longer supported.  (SA1019)
nats_rpc.go:202:9: rpc.ec.Publish is deprecated: Encoded connections are no longer supported.  (SA1019)
nats_rpc_test.go:26:11: nats.DefaultOptions is deprecated: Use GetDefaultOptions() instead. DefaultOptions is not safe for use by multiple clients. For details see #308.  (SA1019)
The command "if [[ "$TRAVIS_GO_VERSION" =~ 1.24 ]]; then find . -type f -name "*.go" | xargs misspell -error -locale US; staticcheck ./...; fi" failed and exited with 1 during .
```
